tests/integration/chroot.py

```diff
@@ -0,0 +1,298 @@
+        image_archive = os.path.exists(self.config["image"])
+        if image_archive:
+            logger.info("Image archive is present: {path}".format(
+              path=self.config["image"]))
+        else:
+            logger.error("Image archive is not present: {path}".format(
+              path=self.config["image"]))
+
+        # Check if port is defined
+        if not "port" in self.config:
+            logger.error("No port for ssh connection defined.")
+        else:
+            logger.info("Port for ssh connection defined: {port}".format(
+              port=self.config["port"]))
+
+        # Check if sshd port can be used
+        self._port_val(self.config["ip"], self.config["port"])
+
+
+    def _unarchive_image(self):
+        """ Unarchive image tar ball """
+        image = self.config["image"]
+        logger.info("Creating tmp_dir")
+        tmp_dir = tempfile.mkdtemp(prefix="gl-int-chroot-")
+        logger.info("Created tmp_dir in {tmp_dir}".format(
+          tmp_dir=tmp_dir))
+        logger.info("Unarchiving image {image} to {tmp_dir}".format(
+          image=image, tmp_dir=tmp_dir))
+        try:
+            with lzma.open(self.config["image"]) as f:
+                with tarfile.open(fileobj=f) as tar:
+                    content = tar.extractall(tmp_dir)
+        except IOError:
+            logger.error("Could not unarchive image file due to IO Error.")
+        except lzma.LZMAError:
+            logger.error("LZMA decompression error.")
+        except tarfile.ReadError:
+            logger.error(("Archive is present but can not be handled by tar."+
+                          "Validate that your tar archive is not corrupt."))
+        except tarfile.CompressionError:
+            logger.error("Used compression is not supported.")
+        except tarfile.HeaderError:
+            logger.error("Malformed tar header information.")
+        logger.info("Unarchived image {image} to {tmp_dir}".format(
+          image=image, tmp_dir=tmp_dir))
+        return tmp_dir
+
+
+    def _generate_ssh_key(self):
+        """ Generate new SSH key for integration test """
+        logger.info("Generating new SSH key for integration tests.")
+        ssh_key_path = self.config["ssh"]["ssh_key_filepath"]
+        # Private key
+        key = paramiko.RSAKey.generate(2048)
+        key.write_private_key_file(ssh_key_path)
+        # Public key (as authorized_keys)
+        public_key = key.get_base64()
+        with open("/tmp/authorized_keys", "w") as f:
+            f.write("ssh-rsa " + public_key)
+        logger.info("SSH key for integration tests generated.")
+
+
+    def _adjust_chroot(self, tmp_dir):
+        """ Adjust chroot to make it usable """
+        logger.info("Adjusting chroot environment")
+        # Adjust garden-epoch to dev
+        chroot_epoch = tmp_dir + "/" + "garden-epoch"
+        chroot_epoch_bool = os.path.exists(chroot_epoch)
+        if chroot_epoch_bool:
```

Comment:
`chroot_epoch_bool` is only used in this if statement. If there is no other need, I would probably define it like this:
```
if os.path.exists(chroot_epoch):
```

Reply:
ok